File: exercism/csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException();
        }

        var distance =
            firstStrand.Zip(secondStrand, (first, second) => first != second)
            .Where(isDifferent => isDifferent)
            .Count();

        return distance;
    }
}

File: exercism/python/hamming/hamming.py
from functools import reduce


def distance(first_strand: str, second_strand: str) -> int:
    if len(first_strand) != len(second_strand):
        raise ValueError('Different length of strands.')

    count_mutation = lambda m, t: m + 1 if t[0] != t[1] else m
    return reduce(count_mutation, zip(first_strand, second_strand), 0)
